### PR TITLE
Validate FC mustStartAtOrAfter + duration

### PR DESCRIPTION
**Pull request by tomquirk**
## What does this PR do and why?

Closes https://github.com/jbx-protocol/juice-interface/issues/1420

Part of remediation for a contract bug. The specific remediation recommendation was:

```
- Web clients should prevent projects from being launched or reconfigured with `_mustStartAtOrAfter + _duration > type(uint54).max. This prevents the proposed start time and that of the upcoming cycle from overflowing a uint54, which has a max value of 2ˆ54 - 1.
```

This PR implements the above.

source: https://juicebox.notion.site/Code4rena-discoveries-d036b810eade4bec98523e01bdf62752

Discord discussion: https://discord.com/channels/775859454780244028/997096039758708776/997441281506676856

## Screenshots or screen recordings

No user-facing changes.

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [x] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by tomquirk** [comment]
Hey there @peripheralist could you please review this when you get a chance?

File: src/constants/numbers.ts [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+
+export const TEN_THOUSAND = 10000
+export const ONE_MILLION = 1000000
+export const ONE_BILLION = 1000000000
+
+export const MaxUint232 = constants.MaxUint256.add(1)
+  .div(2 ** 24)
+  .sub(1)
+export const MaxUint88 = 2 ** 88 - 1
+export const MaxUint54 = BigNumber.from(10).pow(54).sub(1)
```
Comment (wraeth-eth) [inline src/constants/numbers.ts:19]: What is different about this one?

File: src/utils/v2/__tests__/fundingCycle.test.ts [hunk 2]
```diff
@@ -122,4 +126,22 @@ describe('fundingCycle', () => {
       },
     )
   })
+
+  describe('isValidMustStartAtOrAfter', () => {
```
Comment (wraeth-eth) [inline src/utils/v2/__tests__/fundingCycle.test.ts:130]: LFG!!!!!!

File: src/constants/numbers.ts (hunk 1 above)
Comment (tomquirk) [reply src/constants/numbers.ts:19]: it just has `BigNumber` type. `BigNumber.from(2**55 - 1)` is a runtime exception 😢 .

I know its bad for consistency, but... eh

**Comment by tomquirk** [comment]
@johnnyd-eth can you please review this?

File: src/constants/numbers.ts (hunk 1 above)
Comment (johnnyd-eth) [reply src/constants/numbers.ts:19]: Shouldn't it be `BigNumber.from(2).pow(54).sub(1)` ?

**Comment by johnnyd-eth** [comment]
Nice @aeolianeth LGTM, just the one comment about the `MaxUint54`

**Review by johnnyd-eth — CHANGES_REQUESTED** [review]
.

File: src/constants/numbers.ts (hunk 1 above)
Comment (tomquirk) [reply src/constants/numbers.ts]: derp. yes. changed. thank you 🙏

**Comment by tomquirk** [comment]
Thanks @johnnyd-eth comment addressed 🙏

**Review by johnnyd-eth — APPROVED** [review]
Sweet LGTM

**Pull request merged**